packages/core/src/vite-plugin-astro-icon.ts

```diff
@@ -1,153 +1,76 @@
 import type { AstroConfig, AstroIntegrationLogger } from "astro";
-import { mkdir, readFile, writeFile } from "node:fs/promises";
 import type { Plugin } from "vite";
-import type {
-  AstroIconCollectionMap,
-  IconCollection,
-  IntegrationOptions,
-} from "../typings/integration";
-import loadLocalCollection from "./loaders/loadLocalCollection.js";
-import loadIconifyCollections from "./loaders/loadIconifyCollections.js";
-import { createHash } from "node:crypto";
+import { makeSvgComponent } from "astro/assets/utils";
 
-interface PluginContext extends Pick<AstroConfig, "root" | "output"> {
+import { getIconData } from "./utils/icon.js";
+import { FileCache } from "./utils/cache.js";
+import { AstroIconError } from "./utils/error.js";
+
+interface PluginOptions extends Pick<AstroConfig, "cacheDir"> {
   logger: AstroIntegrationLogger;
+  __DEV__: boolean;
 }
 
-export function createPlugin(
-  { include = {}, iconDir = "src/icons", svgoOptions }: IntegrationOptions,
-  ctx: PluginContext,
+export function createVitePlugin(
+  name: string,
+  virtualModulePrefix: string,
+  { cacheDir, logger, __DEV__ }: PluginOptions,
 ): Plugin {
-  let collections: AstroIconCollectionMap | undefined;
-  const { root } = ctx;
-  const virtualModuleId = "virtual:astro-icon";
-  const resolvedVirtualModuleId = "\0" + virtualModuleId;
+  const DEFAULT_ICON_SIZE = 24;
+  const VIRTUAL_MODULE_ID = virtualModulePrefix;
+  const RESOLVED_VIRTUAL_ICON_MODULE_ID = "\0" + VIRTUAL_MODULE_ID;
+
+  const cache = new FileCache(new URL(`${name}/icons/`, cacheDir), logger);
 
   return {
-    name: "astro-icon",
+    name: `vite-plugin-${name}`,
     resolveId(id) {
-      if (id === virtualModuleId) {
-        return resolvedVirtualModuleId;
+      if (id.startsWith(VIRTUAL_MODULE_ID)) {
+        return `\0${id}`;
       }
     },
-
     async load(id) {
-      if (id === resolvedVirtualModuleId) {
+      if (id.startsWith(RESOLVED_VIRTUAL_ICON_MODULE_ID)) {
+        const isRaw = id.endsWith("?raw");
+        const name = id
+          .slice(RESOLVED_VIRTUAL_ICON_MODULE_ID.length)
+          .replace("?raw", "");
+        const [collection, icon] = name.split("/");
+
         try {
-          if (!collections) {
-            collections = await loadIconifyCollections({ root, include });
+          const data = await getIconData(collection, icon, {
+            cache,
+            logger,
+            __DEV__,
+          });
+          if (!data) return;
+
+          const {
+            width = DEFAULT_ICON_SIZE,
+            height = DEFAULT_ICON_SIZE,
+            body,
+          } = data;
+          const svg = `<svg data-icon="${collection}:${icon}" width="${width}" height="${height}" viewBox="0 0 ${width} ${height}">${body}</svg>`;
```

Comment:
Should we switch this to `{collection}/{icon}`? This would match the import rather than having a separate way of referencing the combination

```suggestion
          const svg = `<svg data-icon="${collection}/${icon}" width="${width}" height="${height}" viewBox="0 0 ${width} ${height}">${body}</svg>`;
```